float32 x
float32 y_top
float32 y_center
float32 y_bottom
